examples/features/detached/detached_3_torch_distributed.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# To run:
+# python -m torch.distributed.run --nnodes=1 --nproc_per_node=2 \
+#   --master_addr 127.0.0.1 --master_port 29400 --max_restarts 0 \
+#   detached_3_torch_distributed.py
+
+import logging
+import random
+
+import determined as det
+from determined.experimental import Determined
+
+import torch.distributed as dist
+
+config_text = """
+name: detached-mode-stage-3
+entrypoint: does not matter
+
+checkpoint_storage:
+  host_path: /tmp
+  storage_path: determined-cp
+  type: shared_fs
+
+searcher:
+   name: single
+   metric: loss
+   max_length: 1
+"""
+
+
+def main():
+    logging.basicConfig(format=det.LOG_FORMAT)
+    logging.getLogger("determined").setLevel(logging.INFO)
+    dist.init_process_group("gloo")
+
+    client = Determined()
+
+    if dist.get_rank() == 0:
+        exp_id = det.experimental.detached.create_unmanaged_experiment(client, config_text=config_text)
+        print(f"Created experiment {exp_id}")
+
+        hparams = {"hparam": "distributed"}
+
+        detached_info = det.experimental.detached.create_unmanaged_trial_cluster_info(
+            client, config_text, exp_id, hparams=hparams)
+        broadcast = [exp_id, detached_info]
+    else:
+        broadcast = [None, None]
+
+    # Sync the detached cluster info from the chief.
+    dist.broadcast_object_list(object_list=broadcast, src=0)
+    exp_id, detached_info = broadcast
```

Comment:
I think this part, it would be neat if we helped users do this.  I'm thinking if you shuffle some ordering around, you could offer something like this:
```python
# this is clearly the user's job in detached mode
dist.init_process_group("mybackend")

# still necessary for us
distributed = det.core.DistributedContext.from_torch_distributed()

# should this be incorporated into detached.init()?
exp_id = det.experimental.detached.create_unmanaged_experiment(distributed, ...)

# should this be incorporated into detached.init()?
detached_info = det.experimental.detached.create_unmanaged_trial_cluster_info(distributed, ...)

with det.experimental.detached.init(detached_info, distributed) as core_context:
    ...
```